### PR TITLE
Issue 51

### PR DESCRIPTION
**Pull request by japutko**
Functionality implemented:

- Export participants to CSV file
- Invite dialog that prepares invitation for given emails and then triggers 'mailto:'  (this should be replaced in the future by backend -> sending email directly through SMTP)
- Participants are displayed as static mocks for now -> "join event" functionality needs to be implemented first in order to get real data

### CODE REVIEW

File: client/src/app/modules/events/participant-list/invite-dialog/invite-dialog.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+import { NgbModal, NgbModule, NgbModalRef } from '@ng-bootstrap/ng-bootstrap';
+import { HttpClient } from '@angular/common/http';
+import { TranslateModule, TranslateLoader, TranslateService } from '@ngx-translate/core';
+import { TagInputModule } from 'ngx-chips';
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { InviteDialogComponent } from './invite-dialog.component';
+import { FormsModule, ReactiveFormsModule } from '@angular/forms';
+import { HttpLoaderFactory } from '../../../../app.module';
+import { IEventSchema } from '../../../../../../../server/src/interfaces/mongo.interface';
```
Comment (waaana) [inline client/src/app/modules/events/participant-list/invite-dialog/invite-dialog.component.spec.ts]: Not sure if this is the right way how to do it. We are handling 2 projects in 1 respository. And we can have different model on client side (as we decide not to send other properties of the object).

I'd prefer to create a model in client as well.

File: client/src/app/modules/events/participant-list/participant-list.component.ts [hunk 2]
```diff
@@ -1,29 +1,98 @@
...
+    return person.firstName + ParticipantListComponent.SEPARATOR_WORD +
+           person.lastName + ParticipantListComponent.SEPARATOR_WORD +
+           person.email + ParticipantListComponent.SEPARATOR_LINE;
+  }
 
   constructor() {}
 
-  ngOnInit() {}
+  ngOnInit(): void {
+    // Dummy array for testing until joining events works
```
Comment (waaana) [inline client/src/app/modules/events/participant-list/participant-list.component.ts]: add a [TODO] so we can find it later on, we may forget it & it'll stay in the app

File: client/src/app/modules/events/participant-list/invite-dialog/invite-dialog.component.spec.ts (hunk 1 above)
Comment (japutko) [reply client/src/app/modules/events/participant-list/invite-dialog/invite-dialog.component.spec.ts]: Autoimport "typo". Thanks for noticing.

File: client/src/app/modules/events/participant-list/invite-dialog/invite-dialog.component.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+      this.modal.close('Invitation(s) sent');
+    }
+  }
+
+  public show() {
+    this.open(this.content);
+  }
+
+  private open(content): void {
+    this.modal = this.modalService.open(content, {centered: true, ariaLabelledBy: 'modal-basic-title'});
```
[the remark this answers is not in the document]
Comment (japutko) [reply client/src/app/modules/events/participant-list/invite-dialog/invite-dialog.component.ts]: Not sure why this happens to you. My build runs smoothly with it. I'll find out if we need it at all and will remove it if not.

File: client/src/app/modules/events/event-edit/event-edit.component.html [hunk 4]
```diff
@@ -156,6 +156,6 @@
     </form>
   </div>
   <div class="col-md-4 pr-md-0 pl-md-0" id="participantList">
-    <app-participant-list [iEvent]="iEvent"></app-participant-list>
+    <app-participant-list [(iEvent)]="iEvent"></app-participant-list>
```
[the remark this answers is not in the document]
Comment (japutko) [reply client/src/app/modules/events/event-edit/event-edit.component.html]: Actually, there is no need for two way data binding here. I'll edit it accordingly

File: client/src/app/modules/events/participant-list/invite-dialog/invite-dialog.component.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (japutko) [reply client/src/app/modules/events/participant-list/invite-dialog/invite-dialog.component.ts]: not needed -> removed

File: client/src/app/modules/events/participant-list/invite-dialog/invite-dialog.component.html [hunk 5]
```diff
@@ -0,0 +1,28 @@
+<ng-template #content>
+  <div class="modal-header">
+    <h4 class="modal-title" [translate]="'INVITE_PEOPLE'"></h4>
+    <button type="button" class="close" aria-label="Close" (click)="modal.dismiss('Cross click')">
+      <span aria-hidden="true">&times;</span>
+    </button>
+  </div>
+  <div class="modal-body">
+    <h6>Insert emails:</h6> 
+    <tag-input [(ngModel)]="emails"
```
[the remark this answers is not in the document]
Comment (japutko) [reply client/src/app/modules/events/participant-list/invite-dialog/invite-dialog.component.html]: [ngx-chips options](https://github.com/mauriciovillalobos/ngx-chips#properties-optional)
I find this library super convenient, it has tons of tweaking options. I just fixed the stuff you mentioned :)

File: client/src/app/modules/events/participant-list/invite-dialog/invite-dialog.component.spec.ts [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+describe('InviteDialogComponent', () => {
+  let component: InviteDialogComponent;
+  let fixture: ComponentFixture<InviteDialogComponent>;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [ InviteDialogComponent ],
+      imports: [
+        FormsModule,
```
Comment (waaana) [inline client/src/app/modules/events/participant-list/invite-dialog/invite-dialog.component.spec.ts]: Later on, when I will have more time, we can discuss the stubmodules- they make tests faster & less dependant on other components/ modules

File: client/src/app/modules/events/participant-list/invite-dialog/invite-dialog.component.ts [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+
+  @Input() iEvent: IEventSchema;
+  @ViewChild('content') content;
+
+  closeResult: string;
+  modal: NgbModalRef;
+  validators;
+  errorMessages;
+
+  private emails: Array<any>;
```
[the remark this answers is not in the document]
Comment (waaana) [reply client/src/app/modules/events/participant-list/invite-dialog/invite-dialog.component.ts]: I dont think we agreed upon this :). We can discuss on meetings

**Pull request merged**